x-pack/platform/plugins/shared/triggers_actions_ui/public/common/expression_items/of.test.tsx

```diff
@@ -80,46 +72,26 @@ describe('of expression', () => {
         }}
       />
     );
-    expect(wrapper.find('[data-test-subj="availableFieldsOptionsComboBox"]'))
-      .toMatchInlineSnapshot(`
-      <EuiComboBox
-        async={false}
-        compressed={false}
-        data-test-subj="availableFieldsOptionsComboBox"
-        fullWidth={true}
-        isClearable={true}
-        isInvalid={false}
-        noSuggestions={false}
-        onChange={[Function]}
-        optionMatcher={[Function]}
-        options={
-          Array [
-            Object {
-              "label": "test2",
-            },
-          ]
-        }
-        placeholder="Select a field"
-        selectedOptions={
-          Array [
-            Object {
-              "label": "test",
-            },
-          ]
-        }
-        singleSelection={
-          Object {
-            "asPlainText": true,
-          }
-        }
-        sortMatchesBy="none"
-      />
-    `);
+
+    // Check that the button shows the correct value
+    waitFor(() => {
```

Comment:
This should be `await screen.findBy` instead of wrapped in a `waitFor(`. Additionally, we do 3 times the same query of `getByTestId('ofExpressionPopover')`. This is 3 times in a row going down the tree looking for the same element.

We should search for it once, save it in a constant and then do the `expect`s and `click`s.